rest-api/README.md

```diff
@@ -0,0 +1,321 @@
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `202: Accepted`    | N/A                   |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+### Errors
+| Status | Description            |
+| ------ | ---------------------- |
+| 400    | Validation error(s)    |
+| 400    | Failed Signature Check |
+| 404    | Thread not found       |
+| 409    | RFQ already exists     |
+| 409    | Thread already exists  |
+
+## Get Quote
+### Endpoint
+`GET /threads/:thread_id/?messageType=quote`
+
+### Response
+| Status             | Body                              |
+| ------------------ | --------------------------------- |
+| `200: OK`          | `{ data: TbdexMessage<Quote>[] }` |
+| `400: Bad Request` | `{ errors: Error[] }`             |
+
+### Response Body
+> [!IMPORTANT]
+> See Quote structure [here](../README.md#quote)
+
+## Submit Order
+
+### Endpoint
+`POST /threads/:thread_id/order`
+
+### Order Request Body
+> [!IMPORTANT]
+> See Order structure [here](../README.md#order)
+
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `202: Accepted`    | N/A                   |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+### Errors
+| Status | Description                                                             |
+| ------ | ----------------------------------------------------------------------- |
+| 400    | Failed Signature Check                                                  |
+| 404    | Thread not found                                                        |
+| 409    | Order already exists                                                    |
+| 409    | Order not allowed (e.g. bc thread was closed or bc no quote exists yet) |
+
+## Submit Close
+
+### Description
+Closes the thread. Indicates that Alice is no longer interested
+
+### Endpoint
+`POST /threads/:thread_id/close`
+
+### Request Body
+> [!IMPORTANT]
+> See Close structure [here](../README.md#close)
+
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `202: Accepted`    | N/A                   |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+### Errors
+| Status | Description            |
+| ------ | ---------------------- |
+| 400    | Failed Signature Check |
+| 404    | Thread not found       |
+| 409    | Close not allowed      |
+
+---
+
+## Get Thread
+
+### Description
+Retrieves the thread specified by ID and messageType
+
+### Authentication
+N/A
+
+### Endpoint
+`GET /threads/:id`
+
+
+### Query Params
+| Param         | Description                   |
+| ------------- | ----------------------------- |
+| `messageType` | filters the messages returned |
+
+
+### Response
+
+| Status             | Body                       |
+| ------------------ | -------------------------- |
+| `200: OK`          | `{ data: TbdexMessage[] }` |
```

Comment:
could i assume that for a completed order, the tbdexmessage array returned may look like 
`[Rfq, Quote, Order, OrderStatus, OrderStatus]`, _in that order_ ?

Reply:
Probably not by default - will add a query param for `sort` so that you can get it in an ordered array